habitat_baselines/il/trainers/pacman_trainer.py

```diff
@@ -0,0 +1,633 @@
+
+    def _eval_checkpoint(
+        self,
+        checkpoint_path: str,
+        writer: TensorboardWriter,
+        checkpoint_index: int = 0,
+    ) -> None:
+        r"""Evaluates a single checkpoint.
+
+        Args:
+            checkpoint_path: path of checkpoint
+            writer: tensorboard writer object for logging to tensorboard
+            checkpoint_index: index of cur checkpoint for logging
+
+        Returns:
+            None
+        """
+        config = self.config
+
+        config.defrost()
+        config.TASK_CONFIG.DATASET.SPLIT = self.config.EVAL.SPLIT
+        config.freeze()
+
+        with habitat.Env(config.TASK_CONFIG) as env:
+            nav_dataset = NavDataset(
+                config,
+                env,
+                self.device,
+            ).decode("rgb")
+
+            nav_dataset = nav_dataset.map(nav_dataset.map_dataset_sample)
+
+            eval_loader = DataLoader(nav_dataset)
+
+            logger.info("eval_loader has {} samples".format(len(nav_dataset)))
+
+            q_vocab_dict, ans_vocab_dict = nav_dataset.get_vocab_dicts()
+
+            model_kwargs = {"q_vocab": q_vocab_dict.word2idx_dict}
+            model = NavPlannerControllerModel(**model_kwargs)
+
+            invalids = []
+
+            state_dict = torch.load(checkpoint_path)
+            model.load_state_dict(state_dict)
+            model.eval().to(self.device)
+
+            results_dir = config.RESULTS_DIR.format(split="val")
+
+            metrics = NavMetric(
+                info={"split": "val"},
+                metric_names=[
+                    "{}_{}".format(y, x)
+                    for x in [10, 30, 50]
+                    for y in [
+                        *["d_{}".format(x) for x in [0, "T", "D", "min"]],
+                        "stop",
+                        "ep_len",
+                    ]
+                ],
+                log_json=os.path.join(config.OUTPUT_LOG_DIR, "eval.json"),
+            )
+
+            for t, batch in enumerate(eval_loader):
+                idx, question, answer, actions, action_length, goal_pos = batch
+
+                metrics_slug = {}
+
+                for i in [10, 30, 50]:
+                    if i > action_length.item():
+                        invalids.append([idx.item(), i])
+                        continue
+
+                    question = question.to(self.device)
+
+                    controller_step = False
+                    planner_hidden = model.planner_nav_rnn.init_hidden(1)
+                    # get hierarchical action history
+                    (
+                        planner_actions_in,
+                        planner_img_feats,
+                        controller_step,
+                        controller_action_in,
+                        controller_img_feats,
+                        init_pos,
+                        controller_action_counter,
+                    ) = nav_dataset.get_hierarchical_features_till_spawn(
```

Comment:
@dhruvbatra Should we keep the walked along the shortest path thing?  It is what's in the paper(s) and we never proposed something else concrete, but we do know it's a problematic procedure.

Reply:
For this, @dhruvbatra suggested me to add two more baselines - 

(i) forward-action only - as a fair baseline to the current setting.
(ii) init agent from random position and heading - as a more legit overall performance metric.

I am yet to add those.